? config_office/autom4te.cache
? config_office/config.log
? config_office/config.parms
? config_office/config.status
? config_office/set_soenv
? config_office/warn
? external/glibc/patchinf
? external/glibc/rtufiles/config.h
? external/glibc/rtufiles/getopt.c
? external/glibc/rtufiles/getopt.h
? external/glibc/rtufiles/getopt1.c
? external/glibc/rtufiles/makefile.mk
? external/glibc/rtufiles/readdir_r.c
Index: config_office/configure.in
===================================================================
RCS file: /cvs/tools/config_office/configure.in,v
retrieving revision 1.139
diff -u -u -r1.139 configure.in
--- config_office/configure.in	25 Oct 2005 11:23:57 -0000	1.139
+++ config_office/configure.in	23 Nov 2005 19:07:44 -0000
@@ -226,9 +226,9 @@
 AC_ARG_WITH(system-python,
 [  --with-system-python    Use python already on system
 ],,)
-#AC_ARG_WITH(system-icu,
-#[  --with-system-icu       Use icu already on system
-#],,)
+AC_ARG_WITH(system-icu,
+[  --with-system-icu       Use icu already on system
+],,)
 AC_ARG_WITH(system-db,
 [  --with-system-db        Use berkeley db already on system
 ],,)
@@ -2956,32 +2956,32 @@
 fi
 AC_SUBST(SYSTEM_SANE_HEADER)
 
-# DISABLED; INCOMPLETE
 dnl ===================================================================
 dnl Check for system icu
 dnl ===================================================================
-#AC_MSG_CHECKING([which icu to use])
-#if test -n "$with_system_icu" -o -n "$with_system_libs" && \
-#	test "$with_system_icu" != "no"; then
-#   AC_MSG_RESULT([external])
-#   SYSTEM_ICU=YES
-#   AC_MSG_CHECKING([for unicode/rbbi.h])
-#   AC_TRY_CPP(unicode/rbbi.h, AC_MSG_RESULT([checked.]), AC_MSG_ERROR([icu headers not found.]))
-#   AC_PATH_PROG(SYSTEM_GENBRK, genbrk)
-#   if test -z "$SYSTEM_GENBRK"; then
-#      AC_MSG_ERROR([\"genbrk\" not found in \$PATH, install the icu development tool \"genbrk"\])
-#   fi
-#   AC_PATH_PROG(SYSTEM_GENCCODE, genccode)
-#   if test -z "$SYSTEM_GENCCODE"; then
-#      AC_MSG_ERROR([\"genccode\" not found in \$PATH, install the icu development tool \"genccode"\])
-#   fi
-#else
-#    AC_MSG_RESULT([internal])
-#    SYSTEM_ICU=NO
-#fi
-#AC_SUBST(SYSTEM_ICU)
-#AC_SUBST(SYSTEM_GENBRK)
-#AC_SUBST(SYSTEM_GENCCODE)
+AC_MSG_CHECKING([which icu to use])
+if test -n "$with_system_icu" -o -n "$with_system_libs" && \
+	test "$with_system_icu" != "no"; then
+   AC_MSG_RESULT([external])
+   SYSTEM_ICU=YES
+   AC_MSG_CHECKING([for unicode/rbbi.h])
+   AC_TRY_CPP(unicode/rbbi.h, AC_MSG_RESULT([checked.]), AC_MSG_ERROR([icu headers not found.]))
+   AC_PATH_PROG(SYSTEM_GENBRK, genbrk)
+   if test -z "$SYSTEM_GENBRK"; then
+      AC_MSG_ERROR([\"genbrk\" not found in \$PATH, install the icu development tool \"genbrk"\])
+   fi
+   AC_PATH_PROG(SYSTEM_GENCCODE, genccode)
+   if test -z "$SYSTEM_GENCCODE"; then
+      AC_MSG_ERROR([\"genccode\" not found in \$PATH, install the icu development tool \"genccode"\])
+   fi
+else
+    AC_MSG_RESULT([internal])
+    SYSTEM_ICU=NO
+    BUILD_TYPE="$BUILD_TYPE ICU"
+fi
+AC_SUBST(SYSTEM_ICU)
+AC_SUBST(SYSTEM_GENBRK)
+AC_SUBST(SYSTEM_GENCCODE)
 
 dnl ===================================================================
 dnl Checks for libraries.
Index: config_office/set_soenv.in
===================================================================
RCS file: /cvs/tools/config_office/set_soenv.in,v
retrieving revision 1.78
diff -u -u -r1.78 set_soenv.in
--- config_office/set_soenv.in	28 Oct 2005 11:59:53 -0000	1.78
+++ config_office/set_soenv.in	23 Nov 2005 19:07:47 -0000
@@ -1745,9 +1745,9 @@
 ToFile( "SYSTEM_PYTHON",     "@SYSTEM_PYTHON@",    "e" );
 ToFile( "PYTHON_CFLAGS",     "@PYTHON_CFLAGS@",    "e" );
 ToFile( "PYTHON_LIBS",       "@PYTHON_LIBS@",      "e" );
-#ToFile( "SYSTEM_ICU",        "@SYSTEM_ICU@",       "e" );
-#ToFile( "SYSTEM_GENBRK",     "@SYSTEM_GENBRK@",    "e" );
-#ToFile( "SYSTEM_GENCCODE",   "@SYSTEM_GENCCODE@",  "e" );
+ToFile( "SYSTEM_ICU",        "@SYSTEM_ICU@",       "e" );
+ToFile( "SYSTEM_GENBRK",     "@SYSTEM_GENBRK@",    "e" );
+ToFile( "SYSTEM_GENCCODE",   "@SYSTEM_GENCCODE@",  "e" );
 ToFile( "SYSTEM_JPEG",       "@SYSTEM_JPEG@",      "e" );
 ToFile( "SYSTEM_FREETYPE",   "@SYSTEM_FREETYPE@",  "e" );
 ToFile( "FREETYPE_CFLAGS",   "@FREETYPE_CFLAGS@",  "e" );
Index: i18npool/inc/collator_unicode.hxx
===================================================================
RCS file: /cvs/l10n/i18npool/inc/collator_unicode.hxx,v
retrieving revision 1.8
diff -u -u -r1.8 collator_unicode.hxx
--- i18npool/inc/collator_unicode.hxx	7 Sep 2005 16:51:23 -0000	1.8
+++ i18npool/inc/collator_unicode.hxx	23 Nov 2005 19:08:20 -0000
@@ -84,6 +84,7 @@
     const sal_uInt8 *rulesImage;
 private:
 	RuleBasedCollator *collator;
+	RuleBasedCollator *ucacollator;
 };
 
 #define COLLATOR( algorithm ) \
Index: i18npool/prj/build.lst
===================================================================
RCS file: /cvs/l10n/i18npool/prj/build.lst,v
retrieving revision 1.18
diff -u -u -r1.18 build.lst
--- i18npool/prj/build.lst	1 Nov 2005 14:51:56 -0000	1.18
+++ i18npool/prj/build.lst	23 Nov 2005 19:08:20 -0000
@@ -1,4 +1,4 @@
-inp  i18npool    :   bridges sax stoc comphelper icu i18nutil regexp NULL
+inp  i18npool    :   bridges sax stoc comphelper ICU:icu i18nutil regexp NULL
 inp  i18npool                                   usr1    -   all inp_mkout NULL
 inp  i18npool\source\unotypes                   nmake   -   all inp_utypes NULL
 inp  i18npool\source\registerservices           nmake   -   all inp_rserv inp_utypes NULL
Index: i18npool/source/breakiterator/breakiterator_unicode.cxx
===================================================================
RCS file: /cvs/l10n/i18npool/source/breakiterator/breakiterator_unicode.cxx,v
retrieving revision 1.22
diff -u -u -r1.22 breakiterator_unicode.cxx
--- i18npool/source/breakiterator/breakiterator_unicode.cxx	17 Nov 2005 20:30:35 -0000	1.22
+++ i18npool/source/breakiterator/breakiterator_unicode.cxx	23 Nov 2005 19:08:20 -0000
@@ -42,7 +42,9 @@
 extern const char OpenOffice_icu_dat[];
 U_CDECL_END
 
-using namespace ::com::sun::star::uno;
+using ::com::sun::star::uno::RuntimeException;
+using ::com::sun::star::uno::Reference;
+using ::com::sun::star::uno::Sequence;
 using namespace ::com::sun::star::lang;
 using namespace ::rtl;
 
@@ -103,7 +105,7 @@
             if ( !U_SUCCESS(status) ) throw ERROR;
 
             status = U_ZERO_ERROR;
-            icuBI->aBreakIterator = new RuleBasedBreakIterator(udata_open("OpenOffice", "brk", 
+            icuBI->aBreakIterator = new RuleBasedBreakIterator(udata_open("OpenOffice_icu/OpenOffice", "brk", 
                     OUStringToOString(OUString::createFromAscii(rule)+OUString::createFromAscii("_")+rLocale.Language,
                     RTL_TEXTENCODING_ASCII_US).getStr(), &status), status);
             if (!U_SUCCESS(status) ) {
Index: i18npool/source/breakiterator/makefile.mk
===================================================================
RCS file: /cvs/l10n/i18npool/source/breakiterator/makefile.mk,v
retrieving revision 1.9
diff -u -u -r1.9 makefile.mk
--- i18npool/source/breakiterator/makefile.mk	1 Nov 2005 14:52:19 -0000	1.9
+++ i18npool/source/breakiterator/makefile.mk	23 Nov 2005 19:08:20 -0000
@@ -53,6 +53,8 @@
 # insert "OpenOffice" as icu package name in front of the  name of each rule file for searching on application provided data
 MY_BRK_BRKFILES:=$(subst,data/,$(MISC)$/OpenOffice_ $(MY_BRK_TXTFILES:s/.txt/.brk/))
 
+MY_BRK_NAMEDBRKFILES:=$(subst,data/,OpenOffice_ $(MY_BRK_TXTFILES:s/.txt/.brk/))
+
 # OpenOffice_icu_dat.c is a generated file from the rule file list by gencmn
 MY_MISC_CXXFILES := \
         $(MISC)$/OpenOffice_icu_dat.c \
@@ -80,8 +82,8 @@
 
 # 'gencmn', 'genbrk' and 'genccode' are tools generated and delivered by icu project to process icu breakiterator rules.
 $(MISC)$/OpenOffice_icu_dat.c :  $(MY_BRK_BRKFILES)
-    +$(WRAPCMD) $(SOLARBINDIR)$/gencmn -e OpenOffice_icu -n OpenOffice_icu -S -d $(MISC) O $(mktmp $(MY_BRK_BRKFILES:t"\n"))
+    +$(WRAPCMD) gencmn -e OpenOffice_icu -n OpenOffice_icu -S -d $(MISC) O $(mktmp $(MY_BRK_NAMEDBRKFILES:t"\n"))
 $(MISC)$/OpenOffice_%.brk : data/%.txt
-    +$(WRAPCMD) $(SOLARBINDIR)$/genbrk -r $< -o $(MISC)$/OpenOffice_$*.brk
+    +$(WRAPCMD) genbrk -r $< -o $(MISC)$/OpenOffice_$*.brk
 $(MISC)$/%_brk.c : $(MISC)$/%.brk
-    +$(WRAPCMD) $(SOLARBINDIR)$/genccode -d $(MISC)$ $(MISC)$/$*.brk
+    +$(WRAPCMD) genccode -e OpenOffice_icu -n OpenOffice_icu -d $(MISC)$ $(MISC)$/$*.brk
Index: i18npool/source/collator/collator_unicode.cxx
===================================================================
RCS file: /cvs/l10n/i18npool/source/collator/collator_unicode.cxx,v
retrieving revision 1.9
diff -u -u -r1.9 collator_unicode.cxx
--- i18npool/source/collator/collator_unicode.cxx	17 Oct 2005 15:43:36 -0000	1.9
+++ i18npool/source/collator/collator_unicode.cxx	23 Nov 2005 19:08:20 -0000
@@ -48,12 +48,14 @@
 {
 	implementationName = "com.sun.star.i18n.Collator_Unicode";
 	collator = NULL;
+	ucacollator = NULL;
     rulesImage = NULL;
 }
 
 Collator_Unicode::~Collator_Unicode()
 {
 	if (collator) delete collator;
+	if (ucacollator) delete ucacollator;
 }
 
 sal_Int32 SAL_CALL
@@ -77,7 +79,14 @@
         // load ICU collator
         UErrorCode status = U_ZERO_ERROR;
         if (rulesImage) {
+            ucacollator = (RuleBasedCollator*) icu::Collator::createInstance("", status);
+         if (! U_SUCCESS(status))
+            	throw RuntimeException();
+#ifdef SYSTEM_ICU
+            collator = new RuleBasedCollator(rulesImage, -1, ucacollator, status);
+#else
             collator = new RuleBasedCollator(rulesImage, status);
+#endif
         } else {
            // load ICU collator
             /** ICU collators are loaded using a locale only.
Index: i18npool/source/collator/makefile.mk
===================================================================
RCS file: /cvs/l10n/i18npool/source/collator/makefile.mk,v
retrieving revision 1.9
diff -u -u -r1.9 makefile.mk
--- i18npool/source/collator/makefile.mk	1 Nov 2005 14:53:06 -0000	1.9
+++ i18npool/source/collator/makefile.mk	23 Nov 2005 19:08:20 -0000
@@ -77,6 +77,10 @@
 		$(ICUINLIB) \
 		$(ICUUCLIB)
 
+.IF "$(SYSTEM_ICU)" == "YES"
+CFLAGS += -DSYSTEM_ICU
+.ENDIF
+
 # --- Targets ------------------------------------------------------
 
 .INCLUDE :	target.mk
Index: linguistic/source/hhconvdic.cxx
===================================================================
RCS file: /cvs/sw/linguistic/source/hhconvdic.cxx,v
retrieving revision 1.5
diff -u -u -r1.5 hhconvdic.cxx
--- linguistic/source/hhconvdic.cxx	7 Sep 2005 19:51:22 -0000	1.5
+++ linguistic/source/hhconvdic.cxx	23 Nov 2005 19:08:52 -0000
@@ -34,7 +34,7 @@
  ************************************************************************/
 
 #ifndef USCRIPT_H
-#include <external/unicode/uscript.h>
+#include <unicode/uscript.h>
 #endif
 
 #ifndef _LANG_HXX //autogen wg. LANGUAGE_ENGLISH_US
Index: scp2/prj/build.lst
===================================================================
RCS file: /cvs/installation/scp2/prj/build.lst,v
retrieving revision 1.19
diff -u -u -r1.19 build.lst
--- scp2/prj/build.lst	9 Nov 2005 09:38:56 -0000	1.19
+++ scp2/prj/build.lst	23 Nov 2005 19:08:59 -0000
@@ -1,4 +1,4 @@
-cp    scp2    :    setup_native transex3 PYTHON:python icu NULL
+cp    scp2    :    setup_native transex3 PYTHON:python ICU:icu NULL
 cp    scp2                        usr1     -    all    cp_mkout NULL
 cp    scp2\macros                 nmake    -    all    cp_langmacros NULL
 cp    scp2\source\activex         nmake    -    all    cp_activex cp_langmacros NULL
Index: scp2/source/ooo/file_library_ooo.scp
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/file_library_ooo.scp,v
retrieving revision 1.168
diff -u -u -r1.168 file_library_ooo.scp
--- scp2/source/ooo/file_library_ooo.scp	19 Nov 2005 18:30:27 -0000	1.168
+++ scp2/source/ooo/file_library_ooo.scp	23 Nov 2005 19:08:59 -0000
@@ -877,6 +877,8 @@
     Styles = (PACKED,PATCH);
 End
 
+#ifndef SYSTEM_ICU
+
 File gid_File_Lib_Icudata
     TXT_FILE_BODY;
   #ifdef UNX
@@ -920,6 +922,8 @@
     Dir = gid_Dir_Program;
     Styles = (PACKED);
 End
+
+#endif
 
 File gid_File_Lib_Iiopbrdg
     TXT_FILE_BODY;
Index: scp2/source/ooo/makefile.mk
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/makefile.mk,v
retrieving revision 1.35
diff -u -u -r1.35 makefile.mk
--- scp2/source/ooo/makefile.mk	28 Sep 2005 15:47:30 -0000	1.35
+++ scp2/source/ooo/makefile.mk	23 Nov 2005 19:08:59 -0000
@@ -45,7 +45,6 @@
 # --- Settings -----------------------------------------------------
 
 .INCLUDE :  settings.mk
-.INCLUDE :  icuversion.mk
 
 .IF "$(ENABLE_CRASHDUMP)"!=""
 SCPDEFS+=-DENABLE_CRASHDUMP
@@ -139,10 +138,15 @@
 SCPDEFS+=-DSYSTEM_XALAN
 .ENDIF
 
+.IF "$(SYSTEM_ICU)" == "YES"
+SCPDEFS+=-DSYSTEM_ICU
+.ELSE
+.INCLUDE :  icuversion.mk
 SCPDEFS+=\
     -DICU_MAJOR=$(ICU_MAJOR) \
     -DICU_MINOR=$(ICU_MINOR) \
     -DICU_MICRO=$(ICU_MICRO)
+.ENDIF
 
 SCP_PRODUCT_TYPE=osl
 
Index: scp2/source/ooo/shortcut_ooo.scp
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/shortcut_ooo.scp,v
retrieving revision 1.20
diff -u -u -r1.20 shortcut_ooo.scp
--- scp2/source/ooo/shortcut_ooo.scp	9 Sep 2005 01:42:03 -0000	1.20
+++ scp2/source/ooo/shortcut_ooo.scp	23 Nov 2005 19:08:59 -0000
@@ -167,6 +167,7 @@
 #endif
 
 #ifdef UNX
+#ifndef SYSTEM_ICU
 
 Shortcut gid_Shortcut_Lib_Icudata_0
     FileID = gid_File_Lib_Icudata;
@@ -175,10 +176,6 @@
     Styles = (NETWORK,RELATIVE);
 End
 
-#endif
-
-#ifdef UNX
-
 Shortcut gid_Shortcut_Lib_Icudata_1
     FileID = gid_File_Lib_Icudata;
     Dir = gid_Dir_Program;
@@ -186,10 +183,6 @@
     Styles = (NETWORK,RELATIVE);
 End
 
-#endif
-
-#ifdef UNX
-
 Shortcut gid_Shortcut_Lib_Icui18n_0
     FileID = gid_File_Lib_Icui18n;
     Dir = gid_Dir_Program;
@@ -197,10 +190,6 @@
     Styles = (NETWORK,RELATIVE);
 End
 
-#endif
-
-#ifdef UNX
-
 Shortcut gid_Shortcut_Lib_Icui18n_1
     FileID = gid_File_Lib_Icui18n;
     Dir = gid_Dir_Program;
@@ -208,10 +197,6 @@
     Styles = (NETWORK,RELATIVE);
 End
 
-#endif
-
-#ifdef UNX
-
 Shortcut gid_Shortcut_Lib_Icule_0
     FileID = gid_File_Lib_Icule;
     Dir = gid_Dir_Program;
@@ -219,10 +204,6 @@
     Styles = (NETWORK,RELATIVE);
 End
 
-#endif
-
-#ifdef UNX
-
 Shortcut gid_Shortcut_Lib_Icule_1
     FileID = gid_File_Lib_Icule;
     Dir = gid_Dir_Program;
@@ -230,10 +211,6 @@
     Styles = (NETWORK,RELATIVE);
 End
 
-#endif
-
-#ifdef UNX
-
 Shortcut gid_Shortcut_Lib_Icuuc_0
     FileID = gid_File_Lib_Icuuc;
     Dir = gid_Dir_Program;
@@ -241,10 +218,6 @@
     Styles = (NETWORK,RELATIVE);
 End
 
-#endif
-
-#ifdef UNX
-
 Shortcut gid_Shortcut_Lib_Icuuc_1
     FileID = gid_File_Lib_Icuuc;
     Dir = gid_Dir_Program;
@@ -252,6 +225,7 @@
     Styles = (NETWORK,RELATIVE);
 End
 
+#endif
 #endif
 
 #ifdef UNX
Index: vcl/prj/build.lst
===================================================================
RCS file: /cvs/gsl/vcl/prj/build.lst,v
retrieving revision 1.42
diff -u -u -r1.42 build.lst
--- vcl/prj/build.lst	30 Aug 2005 09:37:21 -0000	1.42
+++ vcl/prj/build.lst	23 Nov 2005 19:09:22 -0000
@@ -1,4 +1,4 @@
-vc	vcl	:	BOOST:boost NAS:nas FREETYPE:freetype psprint rsc sot ucbhelper unotools rvpapi icu i18npool unoil ridljar X11_EXTENSIONS:x11_extensions offuh basegfx SNDFILE:sndfile PORTAUDIO:portaudio transex3 NULL
+vc	vcl	:	BOOST:boost NAS:nas FREETYPE:freetype psprint rsc sot ucbhelper unotools rvpapi ICU:icu i18npool unoil ridljar X11_EXTENSIONS:x11_extensions offuh basegfx SNDFILE:sndfile PORTAUDIO:portaudio transex3 NULL
 vc	vcl										usr1	-	all	vc_mkout NULL
 vc	vcl\source\unotypes						nmake	-	all	vc_unot NULL
 vc	vcl\source\glyphs						nmake	-	all	vc_glyphs vc_unot NULL
